applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_custom_scipy_base_solver.py

```diff
@@ -0,0 +1,198 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Import applications
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+
+# Import base class file
+from KratosMultiphysics.StructuralMechanicsApplication.structural_mechanics_solver import MechanicalSolver
+
+# Import scipy modules
+import KratosMultiphysics.scipy_conversion_tools
+from scipy.sparse.linalg import eigsh
+
+def CreateSolver(main_model_part, custom_settings):
+    return CustomScipyBaseSolver(main_model_part, custom_settings)
+
+class CustomScipyBaseSolver(MechanicalSolver):
+    """The structural mechanics custom scipy base solver.
+
+    This class creates the mechanical solvers to provide mass and stiffness matrices as scipy matrices.
+    
+    Derived class must override the function SolveSolutionStep. In there the Mass and Stiffness matrices
+    can be obtained as scipy matrices.
+    The computation of the eigenvalue problem in this implementation is only an example how this solver is to be used.
+
+    See structural_mechanics_solver.py for more information.
+    """
+    def __init__(self, main_model_part, custom_settings):
+        # Construct the base solver.
+        super(CustomScipyBaseSolver, self).__init__(main_model_part, custom_settings)
+        self.space = KratosMultiphysics.UblasSparseSpace()
```

Comment:
Any particular reason this is a member?
Unless you have one, I would remove it and use it locally
It is the baseclass, derived classes might not need it

Reply:
Well it is used in _MassMatrixComputation and _StiffnessMatrixComputation, so probably also the derived classes will need it, correct?
But I can also declare it as a local variable in both methods?

Reply:
I would make them local variables
Note that the methods in the Space are all `static` hence I think this makes sense

Reply:
Ok will do!